server/routes/profile.go

```diff
@@ -172,6 +173,87 @@ func EditProfile(w http.ResponseWriter, r *http.Request) {
 }
 
 func ProfilePic(w http.ResponseWriter, r *http.Request) {
-	w.WriteHeader(http.StatusAccepted)
-	w.Write([]byte("called pic"))
+	//Connect to datastore
+	ctx := appengine.NewContext(r)
+	dsClient := services.Locator.DsClient()
+
+	//Retrieve user from datastore
+	vars := mux.Vars(r)
+	paramID := vars["id"]
+	userService := services.Locator.UserService()
+	userID := userService.GetCurrentUserID(ctx)
+	if len(userID) == 0 {
+		w.WriteHeader(http.StatusBadRequest)
+		w.Write([]byte("no ID provided"))
+		return
+	}
+	if userID != paramID {
+		w.WriteHeader(http.StatusForbidden)
+		w.Write([]byte("User ID doesn't match"))
+		return
+	}
+
+	k := datastore.NameKey("User", userID, nil)
+
+	var user models.User
+	if err := dsClient.Get(ctx, k, &user); err != nil {
+		log.Printf("Cannot retrieve user from DataStore: %v", err)
+		w.WriteHeader(http.StatusNotFound)
+		w.Write([]byte("entity not found"))
+		return
+	}
+
+	//Get old profile pic
+	oldPic := user.ProfilePic
+
+	//Get new pic
+	file, _, err := r.FormFile("file")
+	if err != nil {
+		serveError(ctx, w, err)
+		return
+	}
+	defer file.Close()
+
+	//Save new pic
+	imageURL, err := UploadImage(ctx, userID, "user_profile_image", nil, file)
+	if err != nil {
+		log.Printf("Error saving image: %v", err)
+		w.WriteHeader(http.StatusInternalServerError)
+		w.Write([]byte("error saving photo"))
+		return
+	}
+
+	//Update user
+	user.ProfilePic = imageURL
+	k, err = dsClient.Put(ctx, k, &user)
+	if err != nil {
+		log.Printf("Cannot save user to DataStore: %v", err)
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	//Delete old pic
+	bucket := config.ImageBucket()
+	image := oldPic[len("https://storage.cloud.google.com//"+bucket):]
+	webURL := os.Getenv("WEBURL")
+	client := &http.Client{}
+	req, err := http.NewRequest("DELETE", webURL+"/images/"+image, nil)
```

Comment:
I strongly recommend not doing an http request to a local server, as it may result in a deadlock in some cases (the servlet is blocking the execution while waiting for the response, but it can't respond as it's blocked). Instead make a function `DeleteImage` and call it both from here and from `HandleDeleteImage`